tests/test_model.py

```diff
@@ -41,6 +41,23 @@ def test_io_nempty(self):
         self.assertTrue(loaded.solve())
         os.remove(fname)
 
+    def test_io_counters(self):
+        fname = join(self.tempdir, "model")
+        iv = cp.intvar(1,9, shape=3)
+        bv = cp.boolvar()
+        m = cp.Model( iv > 3, ~bv )
+        m += (iv[0] == 5)
+        m.to_file(fname)
+
+        self.assertEqual(_BoolVarImpl.counter, 1)
+        self.assertEqual(_IntVarImpl.counter, 3)
+        _BoolVarImpl.counter = 0  # don't try this at home
+        _IntVarImpl.counter = 0  # don't try this at home
+        loaded = cp.Model.from_file(fname)
```

Comment:
But now the user cannot make any variables before loading the model, we should handle this case as well I believe.

Reply:
This is the testcase.

The actual code takes the max of the current and loaded counter, for that reason.